Unity-Technologies/Graphics
# Hdrp/fix hierarchicalbox gizmo symetry and homothety mode ensuring face faces outside

### Purpose of this PR
Better fix for https://fogbugz.unity3d.com/f/cases/1219246/
Previous fix was introducing regression as the reflection probe position was not moving anymore. See discution on https://unity.slack.com/archives/GHD5LADU7/p1594215236202500

---
### Testing status

**Manual Tests**: 
Note. All following test have been done for PlanarReflectionProbe:
- HierarchycalBox center DO NOT move when face collapse in symetry mode (alt key) (all 6 handles)
- HierarchycalBox center DO NOT move when face collapse in homothety mode (shift key) (all 6 handles)
- HierarchycalBox center DO move when manipulating only one face in advanced edition (no key)  (all 6 handles)
- HierarchycalBox center DO NOT move when manipulating only one face but making it collapse to the other end of the box (no key)  (all 6 handles)
- HierarchycalBox center DO NOT move when manipulating it in simple edition mode or with alt+shift key (quickly, not all handles)
- HierarchycalBox size still seams to behave correctly (quickly tested)
- HierarchycalBox still snap with ctrl key (quickly tested)

Cube ReflectionProbe use the same logic so it should work.

I also super quickly tested DensityVolume that also use the HierarchicalBox gizmo.

---
### Comments to reviewers
